### PR TITLE
Add helm post-install hook to automate placement tagging in CouchDB nodes 

### PR DESCRIPTION
**Pull request by fsalazarh**
#### What this PR does / why we need it:

In order to control shard distribution across CouchDB nodes is necessary tagging the nodes in the database `_nodes` with the "zone" attribute. https://docs.couchdb.org/en/stable/cluster/sharding.html#specifying-database-placement

This PR add a `post-install hook` that create a Job that automate the tagging of "zone" attribute, obtaining the info from node labels of kubernetes and the pods allocated on each node. For more information of the procedure itself, please refer to https://github.com/caligrafix/couchdb-autoscaler-placement-manager/blob/main/src/scripts.py

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
  - fixes #

#### Special notes for your reviewer:

We consider that the better way to achieve this is through a configuration file, but currently this solution meets our needs so it's useful. We use this post-install hook in combination with Cluster Autoscaler and Vertical Pod Autoscaler, so this components are responsible for not schedule the pods in other zones after killing them for some reason (e.g. after VPA assigns new requests and limits values). 

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.
- [x] Chart Version bumped
- [x] e2e tests pass
- [x] Variables are documented in the README.md
- [x] Chart tgz added to /docs and index updated

### CODE REVIEW

**Review by kocolosk — CHANGES_REQUESTED** [review]
Thanks for this contribution! I'd like to see if a couple of the added parameters can be eliminated to minimize the chance of user error. But this is a nice convenience.

File: couchdb/README.md [hunk 1]
```diff
@@ -214,6 +214,12 @@ A variety of other parameters are also configurable. See the comments in the
 | `prometheusPort.enabled`             | false                                  |
 | `prometheusPort.port`                | 17896                                  |
 | `prometheusPort.bind_address`        | 0.0.0.0                                |
+| `placementConfig.enabled`            | false                                  |
+| `placementConfig.namespace`          | couchdb                                |
```
[the remark this answers is not in the document]
Comment (fsalazarh) [reply couchdb/README.md:218]: Yes, that is the purpose of the namespace variable in the script. I change this variable to use `{{ .Release.Namespace }}`

File: couchdb/README.md [hunk 2]
```diff
@@ -214,6 +214,12 @@ A variety of other parameters are also configurable. See the comments in the
...
+| `placementConfig.couchdbPort`        | 5984                                   |
```
[the remark this answers is not in the document]
Comment (fsalazarh) [reply couchdb/README.md:219]: In principle no, so I'm going to change it to use `service.externalPort`.

File: couchdb/templates/placementmanager.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+        image: {{ .Values.placementConfig.image.repository }}:{{ .Values.placementConfig.image.tag }}
+        imagePullPolicy: Always
+        args: ["--placement-manager"]
+        envFrom:
+          - secretRef:
+              name: couchdb-couchdb
+          - configMapRef:
+              name: {{ template "couchdb.fullname" . }}
+        env:
+          - name: EKS_NAMESPACE
```
[the remark this answers is not in the document]
Comment (fsalazarh) [reply couchdb/templates/placementmanager.yaml:38]: Done.

File: couchdb/templates/placementmanager.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+            value: {{ .Values.placementConfig.namespace }}
+          - name: COUCHDB_SVC
+            value: {{ template "couchdb.svcname" . }}
+          - name: COUCHDB_PORT
+            value: {{ .Values.placementConfig.couchdbPort | quote }}
+
```
[the remark this answers is not in the document]
Comment (fsalazarh) [reply couchdb/templates/placementmanager.yaml:44]: Done

File: couchdb/values.yaml [hunk 5]
```diff
@@ -221,3 +221,16 @@ sidecars: {}
...
+# Placement manager to annotate each document in the nodes DB with "zone" attribute
+# recording the zone where node has been scheduled
+# Ref: https://docs.couchdb.org/en/stable/cluster/sharding.html#specifying-database-placement
+placementConfig:
+  enabled: false
+  namespace: couchdb # kubernetes namespace 
+  couchdbPort: 5984 # CouchDB Port
+  image:
+    repository: caligrafix/couchdb-autoscaler-placement-manager
+    tag: latest
```
[the remark this answers is not in the document]
Comment (fsalazarh) [reply couchdb/values.yaml:235]: Thanks for this suggestion!, I added a semantic version for placement manager repo.

**Review by fsalazarh — COMMENTED** [review]
I update the code with all suggestions, thanks @kocolosk for review and please review again.

**Comment by kocolosk** [comment]
LGTM, thanks!

**Pull request merged**